packages/mco/components/data-policies/create-dr-policy/create-dr-policy.tsx

```diff
@@ -138,31 +224,131 @@ type SyncScheduleProps = {
   setSchedule: React.Dispatch<React.SetStateAction<string>>;
 };
 
-export const CreateDRPolicy: React.FC<{}> = ({}) => {
+type ReRouteResourceProps = {
+  history: RouteComponentProps['history'];
+  match: Match<{ url: string }>;
+};
+
+export const CreateDRPolicy: React.FC<ReRouteResourceProps> = ({match, history}) => {
   const { t } = useTranslation('plugin__odf-console');
-  const history = useHistory();
+  const { url }  = match;
+  
   const [policyName, setPolicyName] = React.useState('');
-  const [replication] = React.useState('');
-  const [selectedClusters, setSelectedClusters] = React.useState<Cluster[]>([]);
+  const [replication, setReplication] = React.useState('');
+  const [manageS3Profile, setManageS3Profile] = React.useState(false);
   const [syncTime, setSyncTime] = React.useState('5m');
+  const [isODFDetected, setODFDetected] = React.useState(false);
+  const [selectedClusters, setSelectedClusters] = React.useState<ManagedClusterMapping>({});
+  const [isReplicationOpen, setReplicationOpen] = React.useState(false);
+  const [isReplicationInputManual, setIsReplicationInputManual] = React.useState(false);
+  const [errorMessage, setError] = React.useState("");
+  const [mirrorPeers] = useK8sWatchResource<MirrorPeerKind[]>({
+    kind: referenceForModel(MirrorPeerModel),
+    isList: true,
+    namespaced: false,
+  });
+
+  const clustersData =  React.useMemo(() => Object.values(selectedClusters), [selectedClusters]);
+
+  React.useEffect(() => {
+    if (clustersData.length === 2) {
+      // ODF detection
+      setODFDetected(clustersData.every((cluster) => (cluster?.storageSystem !== "" && cluster?.isValidODFVersion)));
+
+      // DR replication type
+      const isReplicationAutoDetectable = clustersData?.every((cluster) => cluster?.storageClusterId !== "");
+      const storageClusterIDs = clustersData?.reduce((ids, cluster) => {
+        if (cluster?.storageClusterId !== "") {
+          ids.add(cluster?.storageClusterId);
+        };
+        return ids;
+      }, new Set());
+      setIsReplicationInputManual(!isReplicationAutoDetectable);
+      isReplicationAutoDetectable ? setReplication(storageClusterIDs.size > 1 ? REPLICATION_TYPE(t)['async'] : REPLICATION_TYPE(t)['sync']) : setReplication(REPLICATION_TYPE(t).async);
```

Comment:
can't we tweak the conditions and combine them ? Just adding an example:
```suggestion
    setReplication(isReplicationAutoDetectable && storageClusterIDs.size  <= 1 ? REPLICATION_TYPE(t)['sync'] : REPLICATION_TYPE(t)['async']) 
```